lib/util/cache/package/file.ts

```diff
@@ -68,3 +68,53 @@ export function init(cacheDir: string): void {
   cacheFileName = upath.join(cacheDir, '/renovate/renovate-cache-v1');
   logger.debug('Initializing Renovate internal cache into ' + cacheFileName);
 }
+
+export async function cleanup(): Promise<void> {
+  logger.debug('Checking file package cache for expired items');
+  try {
+    let totalCount = 0;
+    let deletedCount = 0;
+    for await (const item of cacache.ls.stream(cacheFileName)) {
+      totalCount += 1;
+      const cachedItem = item as unknown as cacache.CacheObject;
```

Comment:
Not sure why I needed to coerce this type

Reply:
strange. we should add another try-catch inside the loop and delete invalid cache values

Reply:
This is like the "cache index" returned by cacache and if it's not the right structure then I'm not sure there's much we can do

Reply:
i mean to delete if JSON parse fails

Reply:
Done now